internal/daemon/controller/handlers/scopes/scope_service.go

```diff
@@ -412,6 +412,26 @@ func (s Service) ListKeys(ctx context.Context, req *pbs.ListKeysRequest) (*pbs.L
 	return &pbs.ListKeysResponse{Items: finalItems}, nil
 }
 
+// RotateKeys implements the interface pbs.ScopeServiceServer.
+func (s Service) RotateKeys(ctx context.Context, req *pbs.RotateKeysRequest) (*pbs.RotateKeysResponse, error) {
+	if req.GetId() == "" {
+		req.Id = scope.Global.String()
+	}
+	if err := validateRotateKeysRequest(req); err != nil {
+		return nil, err
+	}
+	authResults := s.authResult(ctx, req.GetId(), action.RotateScopeKeys)
+	if authResults.Error != nil {
+		return nil, authResults.Error
+	}
+
+	if err := s.kmsRepo.RotateKeys(ctx, req.GetId(), kms.WithRewrap(req.GetRewrap())); err != nil {
+		return nil, err
+	}
+
+	return nil, nil
```

Comment:
I think this may panic the gRPC handles, you want to return an empty message even if there is nothing in it
```suggestion
	return &pbs.RotateKeysResponse{}, nil
```

Reply:
leaving as-is to keep parity with DeleteScope endpoint